From fd6f1cb754d7278dd4eb6e3aeec08898f7e35349 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Mon, 29 Sep 2014 00:59:12 -0400
Subject: [virt] hyperv/net: remove unnecessary null test before kfree

Message-id: <1411952358-34104-29-git-send-email-jasowang@redhat.com>
Patchwork-id: 96768
O-Subject: [RHEL7.1 PATCH 28/34] drivers/net/hyperv/netvsc.c: remove unnecessary null test before kfree
Bugzilla: 1146357
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: Fabian Frederick <fabf@skynet.be>

Bugzilla: 1146357

Fix checkpatch warning:
WARNING: kfree(NULL) is safe this check is probably not required

Cc: Haiyang Zhang <haiyangz@microsoft.com>
Cc: netdev@vger.kernel.org
Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit bd4578bc84a8c8a390cf6002539e75447e78e935)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/hyperv/netvsc.c b/drivers/net/hyperv/netvsc.c
index f7647ef..b96e7f6 100644
--- a/drivers/net/hyperv/netvsc.c
+++ b/drivers/net/hyperv/netvsc.c
@@ -1095,9 +1095,7 @@ close:
  vmbus_close(device->channel);
 
 cleanup:
-
- if (net_device)
-  kfree(net_device);
+ kfree(net_device);
 
  return ret;
 }
-- 
1.7.1